requests/sessions.py

```diff
@@ -623,16 +613,6 @@ def send(self, request, **kwargs):
         stream = kwargs.get('stream')
         hooks = request.hooks
 
-        # Resolve URL in redirect cache, if available.
-        if allow_redirects:
```

Comment:
I hated this code. So glad it's going. 

Hope I'm not responsible for it landing in the codebase. I probably am :)